warehouse/integrations/manager/stats.go

```diff
@@ -0,0 +1,69 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/rudderlabs/rudder-go-kit/stats"
+
+	"github.com/rudderlabs/rudder-server/utils/misc"
+	"github.com/rudderlabs/rudder-server/warehouse/internal/model"
+	warehouseutils "github.com/rudderlabs/rudder-server/warehouse/utils"
+)
+
+type statsManager struct {
+	Manager
+
+	statsFactory stats.Stats
+	stats        struct {
+		addColumnsCount   stats.Counter
+		addTablesCount    stats.Counter
+		alterColumnsCount stats.Counter
+	}
+}
+
+func newStatsManager(manager Manager, statsFactory stats.Stats) *statsManager {
+	return &statsManager{Manager: manager, statsFactory: statsFactory}
+}
+
+func (m *statsManager) Setup(ctx context.Context, warehouse model.Warehouse, uploader warehouseutils.Uploader) error {
+	tags := stats.Tags{
+		"module":      "warehouse",
+		"workspaceId": warehouse.WorkspaceID,
+		"warehouseID": misc.GetTagName(warehouse.Destination.ID, warehouse.Source.Name, warehouse.Destination.Name, warehouse.Source.ID),
```

Comment:
is warehouseID needed? I see all the tags added separately

Reply:
Yeah, it's not needed. keeping for now to be similar to the previous metrics generation.